app/Main.hs

```diff
@@ -187,7 +186,15 @@ requestSecret context secretPath =
             $ setRequestSecure (oConnectTls cliOptions)
             $ defaultRequest
 
-    shouldRetry = const $ return . isLeft
+    -- Only retry on connection related failures
+    shouldRetry _retryStatus (Right _) = return False
+    shouldRetry _retryStatus (Left e) =
+      return $ case e of
+        ServerError _ -> True
+        ServerUnavailable _ -> True
+        ServerUnreachable _ -> True
+        Unspecified _ _ -> True
+        _ -> False
```

Comment:
With a wildcard that defaults to no retry, if somebody adds a new connection-related constructor to `VaultError`, that will not cause a compile error, and the behavior will be to not retry. On the other hand, if we list exceptions, then adding a new error where retries would not help, would still result in retries (if we forget to add the match here). And without wildcards the list will be long, but at least everything is explicit.

I’ll leave it up to you which of the three options is most appropriate here.